src/firefly/js/ui/tap/SpatialSearch.jsx

```diff
@@ -169,14 +178,20 @@ export function SpatialSearch({cols, serviceUrl, columnsModel, tableName, initAr
         return () => setConstraintFragment(panelPrefix, '');
     }, [constraintResult]);
 
-
     return (
         <CollapsibleCheckHeader title={panelTitle} helpID={tapHelpId(panelPrefix)}
-                                message={constraintResult?.simpleError??''}
+                                message={constraintResult?.simpleError ?? ''}
                                 initialStateOpen={true} initialStateChecked={true}>
             <div style={{marginTop: 5}}>
                 <ForceFieldGroupValid forceValid={!checkHeaderCtl.isPanelActive()}>
 
+                    {!canUpload && searchParams.uploadInfo &&
+                        <div>
+                            {<InputFieldLabel label={'Spatial Type:'} tooltip={'tooltip'} labelWidth={SpatialLabelSpatial}/>}
+                            {warningMsg('Warning: Upload is not supported from Single Shape services')}
```

Comment:
Let's improve this message like this:

1. pass `serviceLabel` to spacial search, it could be empty, but most the time not
2. message `Single Shape selected: $(serviceLabel||'This service'} does not support upload`